sklearn/preprocessing/_encoders.py

```diff
@@ -322,6 +322,16 @@ class OneHotEncoder(_BaseEncoder):
         .. versionadded:: 1.1
             Read more in the :ref:`User Guide <one_hot_encoder_infrequent_categories>`.
 
+    feature_name_combiner : "concat_string" or callable, default="concat_string"
+        Callable with signature `def callable(input_feature, category)` that returns a
+        string. This is used to create feature names to be returned by
+        :meth:`get_feature_names_out`.
+
+        `"concat_string"` concatenates encoded feature name and category separated by
+        '_'.  E.g. feature X with values 1, 6, 7 create feature names `X_1, X_6, X_7`.
```

Comment:
I think we can extend this out with a code snippet and state `"concat"` does: `feature + "_" + str(category)`. This way a reader can see that "feature" does not call `str`.